src/Orleans/Messaging/GatewayConnection.cs

```diff
@@ -176,6 +176,17 @@ public void Connect()
             }
         }
 
+        protected override void DrainQueueOnStop(IEnumerable<Message> requests)
```

Comment:
And in this method, instead of doing requestQueue.GetConsumingEnumerable() just loop over  the queue, cal Take one at a time, and resend. If AsynchQueueAgent does not have MsgCenter, just pass it as an arg.